src/pages/fi-pv-expenses/forms/PaymentVoucherExpensesForm.js

```diff
@@ -0,0 +1,739 @@
+    validationSchema: yup.object({
+      accountType: yup.string().required(),
+      currencyId: yup.number().required(),
+      date: yup.string().required(),
+      paymentMethod: yup.string().required(),
+      cashAccountId: yup.string().required(),
+      expenses: yup
+          .array()
+          .of(
+            yup.object().shape({
+              etId: yup.number().required(),
+              amount: yup.number().required(),
+            })
+          )
+          .required()
+    }),
+    onSubmit: async obj => {
+      const copy = { ...obj }
+      delete copy.expenses
+      copy.date = formatDateToApi(copy.date)
+      copy.amount = totalAmount
+      copy.baseAmount = totalAmount
+      const costCentersValues = [];
+
+      const updatedRows = formik.values.expenses.map((expensesDetails, index) => {
+        const { costCenters, ...restDetails } = expensesDetails;
+        if (costCenters) {
+          costCentersValues.push(...costCenters);
+        }
+
+        return {
+          ...restDetails,
+          seqNo: index + 1,
+          pvId: formik.values.recordId || 0,
+        };
+      });
+
+      const data = {
+        header: copy,
+        items: updatedRows,
+        costCenters: costCentersValues,
+      }
+      try {
+        const response = await postRequest({
+            extension: FinancialRepository.PaymentVouchers.set2,
+            record: JSON.stringify(data)
+        })
+
+        !recordId ? toast.success(platformLabels.Added) : toast.success(platformLabels.Edited)
+        const res2 = await getPaymentVouchers(response.recordId)
+        res2.record.date = formatDateFromApi(res2.record.date)
+        getExpenses(res2.record)
+        invalidate()
+      } catch (error) {}
+    }
+  })
+
+
+  const totalAmount = formik.values?.expenses?.reduce((amount, row) => {
+    const amountValue = parseFloat(row.amount?.toString().replace(/,/g, '')) || 0
+
+    return amount + amountValue
+  }, 0)
+
+  const isPosted = formik.values.status === 3
+  const isCancelled = formik.values.status === -1
+  const editMode = !!formik.values.recordId;
+
+  async function getPaymentVouchers(recordId) {
+    try {
+      return await getRequest({
+        extension: FinancialRepository.PaymentVouchers.get,
+        parameters: `_recordId=${recordId}`
+      })
+    } catch (error) {}
+  }
+
+  const onPost = async () => {
+    try {
+      const res = await postRequest({
+          extension: FinancialRepository.PaymentVouchers.post,
+          record: JSON.stringify(formik.values)
+      })
+
+      if (res?.recordId) {
+          toast.success(platformLabels.Posted)
+          invalidate()
+          const res2 = await getPaymentVouchers(res.recordId)
+          res2.record.date = formatDateFromApi(res2.record.date)
+          getExpenses(res2.record)
+      }
+    } catch (exception) {}
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
```

Comment:
replace by const res = await getPaymentVouchers(recordId) since we have it already